src/test/java/jewellery/inventory/unit/service/ProductServiceTest.java

```diff
@@ -0,0 +1,264 @@
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    user.setResourcesOwned(List.of(resourceInUser));
+
+    ProductResponseDto response = new ProductResponseDto();
+    when(productMapper.mapToProductResponseDto(any())).thenReturn(response);
+
+    ProductResponseDto actual = productService.createProduct(productRequestDto);
+
+    assertEquals(actual, response);
+  }
+
+  @Test
+  void testCreateProductShouldRemoveResourceFromUserWhenResourceInUserIsZero() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    resourceInUser.setQuantity(5);
+    user.setResourcesOwned(List.of(resourceInUser));
+    resourceQuantityRequestDto.setQuantity(5);
+
+    productService.createProduct(productRequestDto);
+
+    assertEquals(0, resourceInUserRepository.count());
+  }
+
+  @Test
+  void testCreateProductShouldThrowExceptionWhenResourceInUserNotFound() {
+    user.setResourcesOwned(new ArrayList<>());
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+
+    assertThrows(
+        ResourceInUserNotFoundException.class,
+        () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testCreateProductShouldThrowWhenResourceNotExistInResourceInUserRepository() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    assertThrows(
+        ResourceInUserNotFoundException.class,
+        () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testCreateProductShouldThrowExceptionWhenProductToContainNotFound() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    user.setResourcesOwned(List.of(resourceInUser));
+    productRequestDto.setProductsContent(List.of(testProduct.getId()));
+
+    assertThrows(
+        ProductNotFoundException.class, () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testCreateProductShouldSetContentProduct() {
+
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    user.setResourcesOwned(List.of(resourceInUser));
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    productRequestDto.setProductsContent(List.of(testProduct.getId()));
+
+    ProductResponseDto response = new ProductResponseDto();
+    when(productMapper.mapToProductResponseDto(any())).thenReturn(response);
+
+    ProductResponseDto actual = productService.createProduct(productRequestDto);
+
+    assertEquals(response, actual);
+  }
+
+  @Test
+  void testCreateProductShouldThrowWhenProductNotFound() {
```

Comment:
rename to `testCreateProductShouldThrowWhenContentProductNotFound`. Also this seems to be the same as `testCreateProductShouldThrowExceptionWhenProductToContainNotFound`. Remove one of them.

Reply:
Done!